elasticdl/python/allreduce/pytorch_controller.py

```diff
@@ -75,4 +81,26 @@ def restore(self):
         time.sleep(3)
         # Call `load_state_dict` to reset the state of Horovod optimizer
         self._optimizer.load_state_dict(self._optimizer.state_dict())
+        self._optimizer.zero_grad()
         self._rendezvous_manager.init_horovod_if_needed()
+
+    def reset_backward_passes_per_step(self):
+        world_size = hvd.size()
+        rank = hvd.rank()
+        self.backward_passes_per_step = int(
+            self.batch_num_per_step / world_size
+        )
+        if rank < self.batch_num_per_step % world_size:
+            self.backward_passes_per_step += 1
+        if (
+            self.backward_passes_per_step
+            != self._optimizer.backward_passes_per_step
+        ):
+            self._optimizer.backward_passes_per_step = (
```

Comment:
Will the internal optimizer such as SGD/Adam have the property `backward_passes_per_step`?

Reply:
Yes. The class of optimizer is the subclass of `SGD/Adam` and has attributes of `_ElasticDistributedOptimizer`.  
```python
cls = type(
        optimizer.__class__.__name__,
        (optimizer.__class__,),
        dict(_ElasticDistributedOptimizer.__dict__),
    )
```
The class generated by the code will have attributes and methods of `_ElasticDistributedOptimizer`.

Reply:
I see. I thought it's the internal optimzer inside _ElasticDistributedOptimizer. My mistake.